lib/puppet/provider/package/yum.rb

```diff
@@ -64,15 +65,20 @@ def self.latest_package_version(package, enablerepo, disablerepo, disableexclude
   # @param enablerepo [Array<String>] A list of repositories to enable for this query
   # @param disablerepo [Array<String>] A list of repositories to disable for this query
   # @param disableexcludes [Array<String>] A list of repository excludes to disable for this query
+  # @param environment [Hash<String,String>] An optional hash of environment variables to apply
   # @return [Hash<String, Array<Hash<String, String>>>] All packages that were
   #   found with a list of found versions for each package.
-  def self.check_updates(enablerepo, disablerepo, disableexcludes)
+  def self.check_updates(enablerepo, disablerepo, disableexcludes, environment=nil)
     args = [command(:cmd), 'check-update']
     args.concat(enablerepo.map { |repo| ["--enablerepo=#{repo}"] }.flatten)
     args.concat(disablerepo.map { |repo| ["--disablerepo=#{repo}"] }.flatten)
     args.concat(disableexcludes.map { |repo| ["--disableexcludes=#{repo}"] }.flatten)
 
-    output = Puppet::Util::Execution.execute(args, :failonfail => false, :combine => false)
+    if environment.nil?
```

Comment:
More to the point, I'm still not sure if it will always in the future be the same - for example, if the default value for `custom_environment` gets changed for some reason.